src/person.h

```diff
@@ -263,14 +289,100 @@ class Mechanic : public Person {
 	void Assign(RideInstance *ri);
 	void NotifyRideDeletion(const RideInstance *ri);
 
-	bool DailyUpdate() override;
-	void DecideMoveDirection() override;
-	AnimateResult EdgeOfWorldOnAnimate() override;
 	AnimateResult VisitRideOnAnimate(RideInstance *ri, TileEdge exit_edge) override;
 	RideVisitDesire WantToVisit(const RideInstance *ri, const XYZPoint16 &ride_pos, TileEdge exit_edge) override;
 	void ActionAnimationCallback() override;
 
-	RideInstance *ride;  ///< The ride we're intending to inspect, if any.
+	const Money &GetSalary() const override
+	{
+		return SALARY;
+	}
+};
+
+/**
+ * A guard who walks around the park to stop guests from smashing up park property.
+ * @todo Vandalism is not implemented yet, so the guards don't actually have a function currently.
+ */
+class Guard : public StaffMember {
+public:
+	static const Money SALARY;
+
+	Guard();
+	~Guard();
+
+	void Load(Loader &ldr);
+	void Save(Saver &svr);
+
+	void ActionAnimationCallback() override
+	{
+		NOT_REACHED();  // Guards don't have action animations.
+	}
+
+	const Money &GetSalary() const override
+	{
+		return SALARY;
+	}
+};
+
+/**
+ * An entertainer who walks around the park to make guests more cheerful and patient.
+ * @todo Happiness and impatience are not implemented yet, so the entertainers don't actually have a function currently.
+ */
+class Entertainer : public StaffMember {
+public:
+	static const Money SALARY;
+
+	Entertainer();
+	~Entertainer();
+
+	void Load(Loader &ldr);
+	void Save(Saver &svr);
+
+	void ActionAnimationCallback() override
+	{
+		NOT_REACHED();  // Entertainers don't have action animations.
+	}
+
+	const Money &GetSalary() const override
+	{
+		return SALARY;
+	}
+};
+
+/**
+ * A handyman who walks around the park and waters the flowerbeds, empties the litter bins and sweeps the paths.
+ * @todo Implement watering flowerbeds.
+ * @todo Litter bins and dirty paths are not implemented yet.
+ */
+class Handyman : public StaffMember {
+public:
+	static const Money SALARY;
+
+	/** What a handyman is doing right now. */
+	enum Activity {
+		ACTIVITY_NONE,      ///< Wandering around.
+		ACTIVITY_WATER,     ///< Watering a flowerbed.
+		ACTIVITY_SWEEP,     ///< Sweeping the path.
+		ACTIVITY_EMPTY_NE,  ///< Emptying the bin on the north-east edge of the current tile.
+		ACTIVITY_EMPTY_SE,  ///< Emptying the bin on the south-east edge of the current tile.
+		ACTIVITY_EMPTY_SW,  ///< Emptying the bin on the south-west edge of the current tile.
+		ACTIVITY_EMPTY_NW,  ///< Emptying the bin on the north-west edge of the current tile.
+	};
```

Comment:
Something "Handyman" in these names is likely useful, and/or rename the "none" (perhaps "wander"?) activity, as that is very generic. 
Another way to have a 'handyman' in there is to use an enum class, where you are forced to also list the enum name, eg `Activity::ACTIVITY_NONE`. Then you always have 'activity' twice in the current implementation, so there is room for the staff name, or dropping of one activity.
(I'd aim for something like `HandymanAction::WANDER`, I think.)

[enum classes does make merging all activities into one spot, eg the base class, more difficult due to the increased correct typing that is enforced, so it may not fly here?]